boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,201 @@
+    // Invert watchdog bit state
+    if (e_fuse->watch_dog_state)
+    {
+        SET_BIT(command, WATCH_DOG_BIT);
+    }
+    else
+    {
+        CLEAR_BIT(command, WATCH_DOG_BIT);
+    }
+    // Invert watchdog bit state for next write
+    e_fuse->watch_dog_state = !e_fuse->watch_dog_state;
+
+    // Compute and set/clear parity value
+    Io_Efuse_SetParityBit(&command);
+
+        status = Io_Efuse_WriteToEfuse(
+            &command, e_fuse->chip_select.GPIO_Port,
+            e_fuse->chip_select.GPIO_Pin);
+    UNUSED(status);
+}
+
+uint16_t Io_Efuse_ReadReg(uint8_t register_address, struct Efuse *e_fuse)
+{
+    HAL_StatusTypeDef status;
+    uint16_t          command   = 0x0000U;
+    uint16_t          tx_data[] = { 0x0000U };
+    uint16_t          rx_data[] = { 0x0000U };
+
+    // Place the Status Register address into bits 10->13
+    command =
+        (uint16_t)((SI_STATR_0_ADDR & EFUSE_ADDR_MASK) << EFUSE_ADDR_SHIFT);
+    // Shift bit 3 of the address (SOA3: the channel number) to the 13th bit
+    command = (uint16_t)(
+        command | ((register_address & SOA3_MASK) << EFUSE_ADDR_SHIFT) |
+        (register_address & (SOA2_MASK | SOA1_MASK | SOA0_MASK)));
+    // Invert watchdog bit state
+    if (e_fuse->watch_dog_state)
+    {
+        command |= WATCH_DOG_BIT;
+    }
+    else
+    {
+        command &= ~(WATCH_DOG_BIT);
+    }
+    // Invert watchdog bit state for next write
+    e_fuse->watch_dog_state = !e_fuse->watch_dog_state;
+
+    // Compute and set/clear parity bit
+    Io_Efuse_SetParityBit(&command);
+
+    tx_data[0] = command;
+    status     = Io_Efuse_ReadFromEfuse(
+        tx_data, rx_data, e_fuse->chip_select.GPIO_Port,
+        e_fuse->chip_select.GPIO_Pin);
+    UNUSED(status);
+    // Only return register contents and clear bits 9->15
+    CLEAR_BIT(rx_data[0], ~EFUSE_SO_DATA_MASK);
+    return rx_data[0];
+}
+
+void SetParityBit(uint16_t *spi_command)
+{
+    uint16_t spi_data = *spi_command;
+    uint8_t  parity_bit;
+    // Calculate parity for bits in spi command
+    for (parity_bit = 0; spi_data > 0; spi_data >>= 1)
+    {
+        parity_bit ^= (spi_data & 1);
+    }
+
+    if (parity_bit == 1)
+    {
+        SET_BIT(*spi_command, PARITY_BIT);
+    }
+    else
+    {
+        CLEAR_BIT(*spi_command, PARITY_BIT);
+    }
+}
+
+HAL_StatusTypeDef Io_Efuse_WriteToEfuse(
+    uint16_t *    TxData,
+    GPIO_TypeDef *GPIO_Port,
+    uint16_t      GPIO_Pin)
+{
+    HAL_StatusTypeDef status;
+
+    HAL_GPIO_WritePin(GPIO_Port, GPIO_Pin, GPIO_PIN_RESET);
+    status = HAL_SPI_Transmit(efuse_spi_handle, (uint8_t *)TxData, 1U, 100U);
+    HAL_GPIO_WritePin(GPIO_Port, GPIO_Pin, GPIO_PIN_SET);
+
+    return status;
+}
+
+HAL_StatusTypeDef Io_Efuse_ReadFromEfuse(
+    uint16_t *    tx_data,
+    uint16_t *    rx_data,
+    GPIO_TypeDef *chip_select_port,
+    uint16_t      chip_select_pin)
+{
+    HAL_StatusTypeDef transmit_status, receive_status;
```

Comment:
```suggestion
    HAL_StatusTypeDef status;
```
I thought I left a comment about this earlier but not sure. Let's combine the two statuses into one and return it.